### PR TITLE
Refactor API error handling

### PR DESCRIPTION
**Pull request by bastilian**
This rewrites the API client to reject a promise in case an error occurs and trigger the notifications middleware, which will look for title and details in the response to show the error.

Requires https://github.com/RedHatInsights/notifications-backend/pull/54 in the backend

### CODE REVIEW

File: src/SmartComponents/NotificationEdit/NotificationEdit.js [hunk 1]
```diff
@@ -155,7 +155,14 @@ export class NotificationEdit extends Component {
         ((endpoint) ? this.props.updateEndpoint(endpoint.id, payload) : this.props.createEndpoint(payload))
         .then(this.toIndex)
         .catch(() => {
-            const errors = _.mapValues(this.props.endpointErrors.errors, (value) => ({ errors: value }));
+            let errors = {};
+            _.forEach(this.props.endpointErrors, (value) => {
```
Comment (adamruzicka) [inline src/SmartComponents/NotificationEdit/NotificationEdit.js:159]: When I was testing this, no matter what I tried `this.props.endpointErrors` was always undefined.

**Review by adamruzicka — CHANGES_REQUESTED** [review]
Works mostly fine when the backend server is up and returning errors.

The only case where it doesn't work are validation errors. Trying to create an endpoint with a name that is already taken fails with just "Error" notification showing up. Also the fields don't get marked as invalid.

If the server is down, just generic errors are shown.

Also an error is shown when you go to endpoint details and it doesn't have a filter assigned. In this case we can safely use empty object as a filter and not show an error

**Comment by bastilian** [comment]
@adamruzicka Mind taking another look.

Validation errors are now passed on correctly and are marking the fields as they should.
There is also now the option to ignore 404 errors for filters to not show an error at all.

When the backend is down the proxy seems to still respond with 200, and and empty response, I'm unsure how to identify those properly as errors with the backend to show a fitting notification.

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thanks @bastilian !

**Pull request merged**